logstash-core/spec/logstash/pipeline_spec.rb

```diff
@@ -840,6 +843,64 @@ class TestPipeline < LogStash::Pipeline
           expect(collected_metric[:stats][:pipelines][:main][:plugins][:filters][plugin_name][:name].value).to eq(LogStash::Filters::Multiline.config_name)
         end
       end
+
+      context 'when dlq is disabled' do
+        let (:collect_stats) { subject.collect_dlq_stats}
+        let (:collected_stats) { collected_metric[:stats][:pipelines][:main][:dlq]}
+        let (:available_stats) {[:path, :current_queue_size_in_bytes]}
+
+        before :each do
+          logstash_settings = LogStash::SETTINGS.get_setting("dead_letter_queue.enable")
+          allow(logstash_settings).to receive(:value).and_return(false)
+        end
+
+        it 'should show the dlq as disabled' do
+          collect_stats
+          expect(collected_stats[:enabled].value).to be false
+        end
+
+        it 'should not show other stats' do
+          collect_stats
+          available_stats.each do |stat|
+            expect(collected_stats[stat]).to be_nil
+          end
+        end
+      end
+
+      context 'when dlq is enabled' do
+        # Hook to setup mocks to enable a stubbed dlq_writer to be set.
+        let (:before_subject_hook) {
+          allow_any_instance_of(LogStash::BasePipeline).to receive(:dlq_writer).and_return(dlq_writer)
+          allow(dlq_writer).to receive(:get_path).and_return(path)
+          allow(path).to receive(:to_absolute_path).and_return(queue_path)
+          allow(dlq_writer).to receive(:get_current_queue_size).and_return(queue_size)
+          allow(dlq_writer).to receive(:close)
+        }
+
+        let (:queue_size) { 1024 }
+        let (:queue_path) { '/tmp' }
```

Comment:
I don't _think_ it will interfere with other tests, as no entries are written to any directory - this test uses a test double for dlq_writer just to return stats.

I think I have an idea how to do this with a few less egregious mocks...